subsys/profiler/profiler_common_shell.c

```diff
@@ -71,7 +65,8 @@ static void set_event_profiling(const struct shell *shell, size_t argc,
 		}
 
 		for (size_t i = 0; i < index_cnt; i++) {
-			evt_mask |= BIT(event_indexes[i]);
+			atomic_set_bit_to(_profiler_event_enabled_bm.flags,
```

Comment:
Consider making whole operations atomic here. See comments added to Event Manager shell for details.

Reply:
I was thinking mainly about case where shell gets preempted after only some of the flags are changed (it uses low priority preemptive context). In that case we may be in a state where some of the events are already affected by our shell command and some of them are not yet affected. I think it could be confusing to end user.

Reply:
Still, if you think it's not necessary, then we can leave it as it is

Reply:
> In that case we may be in a state where some of the events are already affected by our shell command and some of them are not yet affected. I think it could be confusing to end user.

I understood. Making entire operation atomic will not help. Even if the change happens right after the command completes you will get some bit set differently then user has requested.